### PR TITLE
Added JMX support for microservices instance

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

**Comment by ronenhamias** [comment]
i would suggest to look on zipkin tracing vs jmx

**Comment by segabriel** [comment]
@ronenhamias 
> i would suggest to look on zipkin tracing vs jmx

This PR is not about tracing - we need to have some mechanism to get to know which settings were applied when microservices starts.

File: services/src/main/java/io/scalecube/services/MicroservicesMBean.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  Map<String, InetSocketAddress> getGatewayAddresses();
+
+  String getServiceEndpoint();
+
+  List<String> getRecentServiceDiscoveryEvents();
+
+  List<String> getServiceEndpoints();
+
+  List<String> getServiceReferences();
```
Comment (artem-v) [inline services/src/main/java/io/scalecube/services/MicroservicesMBean.java]: Let's remove this. `getServiceEndpoints()` has it all.

File: services/src/main/java/io/scalecube/services/MicroservicesMBean.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  String getServiceTransport();
+
+  String getServiceDiscovery();
+}
```
Comment (artem-v) [inline services/src/main/java/io/scalecube/services/MicroservicesMBean.java:22]: Let's return even single String as  List<String>   (for conveniet  rendering in jconsole)

**Review by artem-v — DISMISSED** [review]

**Review by artem-v — APPROVED** [review]

**Pull request merged**